### ISSUE TITLE
Should parsing + pretty printing round trip?

### ISSUE THREAD

**Issue by AlistairB** [fosskers/versions#42]
ie this removes the `0`.

```
λ> DV.prettySemVer <$> DV.semver "2.2.1-b05"
Right "2.2.1-b5"
```

My use case is I was to convert a bunch of textual versions to `Version`, find the maximum, then get back the same textual representation for further use.

Not sure if this is a bug as such or a design decision.

**Comment by fosskers** [comment]
They should indeed round-trip. You may have found a bug.

**Comment by fosskers** [comment]
The silver lining is that my [Rust port](https://lib.rs/crates/versions) of this library doesn't suffer from this bug, so I can compare the two and more easily see why.

**Comment by fosskers** [comment]
Here's what we expect the prerelease section to parse as:
```rust
    #[test]
    fn semver_zeros() {
        let pre_rel = Chunks(vec![Chunk(vec![
            Unit::Letters("b".to_string()),
            Unit::Digits(0),
            Unit::Digits(5),
        ])]);

        assert_eq!(
            Some(pre_rel),
            SemVer::new("2.2.1-b05").and_then(|sv| sv.pre_rel)
        );
    }
```

**Comment by fosskers** [comment]
Naughty:
```
      Zero Handling
        2.2.1-b05:                                       FAIL
          test/Test.hs:126:
          expected: Right (SemVer {_svMajor = 2, _svMinor = 2, _svPatch = 1, _svPreRel = [Str "b" :| [Digits 0,Digits 5]], _svMeta = []})
           but got: Right (SemVer {_svMajor = 2, _svMinor = 2, _svPatch = 1, _svPreRel = [Str "b" :| [Digits 5]], _svMeta = []})
```

**Comment by AlistairB** [comment]
Looks like the issue boils down to `Digits` holding a `Word` in `data VUnit = Digits Word | Str Text`. As `Word` cannot have leading zeros. ie. `read "05" :: Word -- 5`.

I suppose you could represent `Digits` as `Digits (NonEmpty Word)` with each `Word` being a single digit. Or alternatively just use test ie. `Digits Text`.

**Comment by fosskers** [comment]
It's just a parsing issue - it should be two separate `Digits` values.

**Comment by AlistairB** [comment]
Oh right I see

**Comment by fosskers** [comment]
Fixed and released in `4.0.2`.

**Comment by AlistairB** [comment]
Thanks! ❤️
